akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -224,38 +226,34 @@ private[http] object StreamUtils {
    * is finished, only that the part that contains this flow has finished work.
    */
   def identityFinishReporter[T]: Flow[T, T, Future[Unit]] = {
-    // copy from Sink.foreach
-    def newForeachStage(): (PushStage[T, T], Future[Unit]) = {
-      val promise = Promise[Unit]()
+    Flow[T].viaMat(new GraphStageWithMaterializedValue[FlowShape[T, T], Future[Unit]] {
```

Comment:
Instead of creating a new GraphStageWithMaterializedValue, make it an `object IdentityFinishReporter extends GraphStageWithMaterializedValue …`